src/accelerate/big_modeling.py

```diff
@@ -327,9 +327,13 @@ def dispatch_model(
     # Error early if the device map is incomplete.
     check_device_map(model, device_map)
 
+    # for backward compatibility
+    is_quantized = getattr(model, "is_quantized", False) or getattr(model, "is_loaded_in_8bit", False)
+
     # We attach hooks if the device_map have at least 2 different devices. Otherwise, the model in already loaded
     # in the unique device and the user can decide where to dispatch the model.
-    if len(device_map.values()) > 1:
+    # If the model is quantized, we always force-dispatch the model
+    if (len(set(device_map.values())) > 1) or is_quantized:
```

Comment:
Just to confirm, if you want to check the number of devices maybe `set(xxx)` was more appropriate? (reading from the comments above)

Reply:
works for me